### DISCUSSION TITLE
REST API Post Parameters (getEvent)

### DISCUSSION THREAD

**Discussion by MarioCipo** [SauceLLC/sauce4zwift#45] [General]
Want to get my current event, which  i am currently attend with REST API. How to fill the request-post parameters to receive a valid response?

POST http://localhost:1080/api/rpc/getEvent
Content-Type: application/json; charset=UTF-8

{
  ????
}

**Comment by mayfield** [comment]
Hi Mario,

getEvent takes a single argument of the event ID,  The post body should be
a JSON array with one number like, e.g.


[3100000]

Probably the easiest way to figure out most of the RPC methods is look at
the code or watch the network tab of a window with devtools (it hit F12).

Justin

**Comment by stuartlynne** [comment]
How do we get the event ID for an event if the rider is in an event?

**Reply by MarioCipo** [reply]
my solution: get EventId with /api/athletes/self, then use the EventSubgroupID with /api/rpc/getEventSubgroupEntrants to get all current participants in the event.
